### ISSUE TITLE
Designated initializer for struct with embedded struct shows missing field

### ISSUE THREAD

**Issue by ghost** [microsoft/vscode-cpptools#2363]
Type: LanguageService
Designated initializer for struct with embedded struct shows missing field but shouldn't.

- OS and Version: Windows 7
- VS Code Version: 1.25.1
- C/C++ Extension Version: 0.17.7
- gcc (i686-posix-dwarf, Built by MinGW-W64 project) 6.1.0

**To Reproduce**
Save following code as c file:
```
struct foo {
    int x;
};

struct bar {
    struct foo;
    int y;
};

int main() {
    struct bar bar = {.x=12, .y=13};
}
```
Shown message:
![struct_embedding](https://user-images.githubusercontent.com/7446828/43759876-e1d61b6e-9a20-11e8-9279-edd75c0f919e.png)

The code compiles and runs though. 

**Expected behavior**
No missing field message shown.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. It's a bug/deficiency with our parser in gcc/clang mode (msvc-mode works, but that assumes you're using cl.exe to compile, so you wouldn't want to use that). I've requested a bug report be filed for this.

**Comment by sean-mcmanus** [comment]
That code doesn't build with the "normal" gcc compilers (see https://gcc.godbolt.org/ ), but it does compile with the mingw64 and strawberryperl...do you happen to know what default flags are being used with that version of gcc that could lead to the different compilation behavior? I assume they're enabling some gcc extension by default? I'm trying to figure out how to get the other gcc compilers to compile that code.

**Comment by sean-mcmanus** [comment]
Figured it out -- the gcc mingw-64 compiler has -fms-extensions enabled by default (using -fno_ms_extensions gets the compiler to fail). I'm still investigating a potential workaround...

**Comment by sean-mcmanus** [comment]
Adding "--ms_extensions" to the msvc.64.intel.clang.json file (next to the binary, i.e. `.vscode\extensions\ms-vscode.cpptools-0.17.7\bin`) is a workaround, but the IntelliSense process will crash if you hover over the .x (due to bug https://github.com/Microsoft/vscode-cpptools/issues/2370, which should be fixed in our next release), so avoid that. Also, you need to make sure the file is being interpreted as "C" file (won't work with C++), i.e. it should say "C" in the bottom right.

e.g. it should look like:
```
      "8",
      "--ms_extensions"
```

**Comment by ghost** [comment]
Thanks for investigating the issue. I wasn't aware that this syntax is not part of the c standard. It's nice that the cpptools might support it in the future anyway.

**Comment by ghost** [comment]
Should I close the issue as it's looks more like a feature request now (support for ms extensions in the gcc compiler)?

**Comment by sean-mcmanus** [comment]
Adding support for -fms-extensions and -fno-ms-extensions args to the compilerPath and respecting mingw's default of -fms-extensions has already been implemented and should be in our next release. You can close the issue if you want or keep it open and we'll close it when we ship 0.17.8.

**Comment by sean-mcmanus** [comment]
Fixed with 0.18.0.
